spec/function.dd

```diff
@@ -1803,6 +1785,53 @@ void venus()
 }
 ---
 
+$(H4 $(LNAME2 struct-return-methods, Struct Return Methods))
+
+        $(P Struct non-static methods marked with the `return` attribute ensure the returned
+        reference will not outlive the struct instance.
+        )
+
+$(SPEC_RUNNABLE_EXAMPLE_FAIL
+---
+struct S
+{
+    private int x;
+    ref int get() return { return x; }
+}
+
+ref int escape()
+{
+    S s;
+    return s.get(); // Error: escaping reference to local variable s
+}
+---
+)
+        $(P The `return` attribute can also be used for wrapper structs to limit
+        the lifetime of the returned value:
+        )
+
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+---
+@safe:
```

Comment:
This is easily missed. Better annotate every function with `@safe` explicitly. It’s just two of them.